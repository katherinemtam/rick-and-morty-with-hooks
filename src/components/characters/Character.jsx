import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image }) => {
  return (
    <>
      <h2>{name}</h2>
      <img src={image} />
    </>
  );
};

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Character;
