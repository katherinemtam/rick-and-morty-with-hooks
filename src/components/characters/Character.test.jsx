import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  const character = {
    name: 'Pickle Rick',
    image: 'http://image.com'
  };

  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<Character name={character.name} image={character.image}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
