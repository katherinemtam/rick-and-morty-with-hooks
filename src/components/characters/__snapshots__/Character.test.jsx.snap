// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <h2>
    Pickle Rick
  </h2>
  <img
    src="http://image.com"
  />
</DocumentFragment>
`;
